blueflood-core/src/main/java/com/rackspacecloud/blueflood/io/datastax/DAbstractPreaggregatedIO.java

```diff
@@ -0,0 +1,333 @@
+/*
+ * Copyright (c) 2016 Rackspace.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * 	http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.rackspacecloud.blueflood.io.datastax;
+
+import com.datastax.driver.core.*;
+import com.google.common.collect.HashBasedTable;
+import com.google.common.collect.Table;
+import com.rackspacecloud.blueflood.io.CassandraModel;
+import com.rackspacecloud.blueflood.io.Instrumentation;
+import com.rackspacecloud.blueflood.rollup.Granularity;
+import com.rackspacecloud.blueflood.types.*;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.lang.Long;
+import java.nio.ByteBuffer;
+import java.util.*;
+
+/**
+ * This is an abstract class that collects the common behavior of
+ * reading/writing preaggregated metrics objects.
+ */
+public abstract class DAbstractPreaggregatedIO {
```

Comment:
`DatastaxAbstractPreaggregatedIO` is not too long. You know what is too long? This guy `SimpleBeanFactoryAwareAspectInstanceFactory`